std/traits.d

```diff
@@ -5723,6 +5724,73 @@ enum bool isAutodecodableString(T) = (is(T : const char[]) || is(T : const wchar
 }
 
 /**
+Checks whether `T` behaves similar to a string. In other words it requires `T`
+to be an $(B finite) `InputRange` with `char`, `wchar` or `dchar`
+as $(REF ElementEncodingType, std, range).
+Moreover `T` isn't allowed to be an aggregate type,
+implicitly convertible enum or static string array.
+
+Params:
+     T = type to be tested
+
+Returns:
+    true if T behaves similar to a string
+
+See_Also:
+    $(LREF isSomeChar), $(LREF isNarrowString), $(LREF StringTypeOf)
+*/
+template isStringLike(T)
```

Comment:
> bikeshedding: IMO the name isn't really clear when looked at in a constraint. How about isStringLikeRange?

Andrei suggested [`isStringLike`](http://forum.dlang.org/post/o827vs$1kt3$1@digitalmars.com), but mentioning "range" in the name isn't a bad idea ;-)